### PR TITLE
smol pr for changing pygresql to use psycopg2 in blip api fr this time

### PR DESCRIPTION
**Pull request by chmnata**
## What this pull request accomplishes:

- Remove the usage of pygresql package and use psycopg2 instead

## Issue(s) this solves:

- #131

## What, in particular, needs to reviewed:

- If it works
- I feel like there was something about the upsert that I had problem with, so the upsert statement
- Now that I think about it more, I think there was a problem with upsert that I haven't fixed yet

### CODE REVIEW

File: bluetooth/api/blip_api.py [hunk 1]
```diff
@@ -165,6 +171,10 @@ def get_wsdl_client(wsdlfile, direct=None, live=False):
             config[key] = zeep.xsd.SkipValue
     return blip, config
 
+class Object:
```
Comment (radumas) [inline bluetooth/api/blip_api.py:174]: I have a suspicion about what this does but it should be documented

File: bluetooth/api/blip_api.py [hunk 2]
```diff
@@ -267,22 +307,23 @@ def main(dbsetting: 'path/to/config.cfg' = None,
...
     blip, config = get_wsdl_client(api_settings['WSDLfile'], direct, live)
 
     try:
-        db = DB(**dbset)
+        conn = _get_db(dbset)
     except InternalError as _:
         LOGGER.error('Connection error to RDS, sleeping for ten minutes')
         sleep(60 * 10)
-        db = DB(**dbset)
+        conn = _get_db(dbset)
```
Comment (radumas) [inline bluetooth/api/blip_api.py:319]: I think this part can be changed to similar to the other _get_db() block

**Review by radumas — CHANGES_REQUESTED** [review]
This should also update the requirements file.

I can't see anything obviously wrong with the upsert query

File: bluetooth/api/blip_api.py (hunk 2 above)
Comment (chmnata) [reply bluetooth/api/blip_api.py:319]: like which ones ? 😵

Comment (radumas) [reply bluetooth/api/blip_api.py:319]: dis
https://github.com/CityofToronto/bdit_data-sources/blob/2e9a12801bfa6382f3c8113824a469961361afa2/bluetooth/api/blip_api.py#L122-L127

File: bluetooth/api/blip_api.py [hunk 3]
```diff
@@ -177,70 +187,100 @@ def update_configs(all_analyses, dbset):
...
+                cur.execute("SELECT king_pilot.load_bt_data();")
+                query = cur.fetchone()    
+        if query[0] != 1:
+            conn.rollback()
+            raise DatabaseError('king_pilot.load_bt_data did not complete successfully') 
+        ## Truncate and delete if successful 
+        with conn:
+            with conn.cursor() as cur:
+                cur.execute("TRUNCATE bluetooth.raw_data;")
+                cur.execute("DELETE FROM king_pilot.daily_raw_bt WHERE measured_timestamp < now()::DATE;")
```
Comment (chmnata) [inline bluetooth/api/blip_api.py:275]: is this `now()::DATE` bad? 👁️ 👁️

Comment (radumas) [reply bluetooth/api/blip_api.py:275]: zomg yes.
Also rather tempted to say we should delete all the `king_pilot` stuff, but mayyyyybe pulling the live data hourly can be useful again.

**Review by radumas — CHANGES_REQUESTED** [review]
I will concede it works. I will reiterate it would be nice to document the subclassing Object to hack the json representation function

File: bluetooth/api/blip_api.py (hunk 2 above)
Comment (radumas) [reply bluetooth/api/blip_api.py]: you have to link to the permalink

image

**Review by radumas — CHANGES_REQUESTED** [review]
requirements.txt needs to be updated.
Does the config file need to be changed?

**Comment by chmnata** [comment]
> requirements.txt needs to be updated. Does the config file need to be changed?

oh yea yes

**Review by radumas — APPROVED** [review]
Party time. :meow_party: Excellent :deployparrot:

**Pull request merged**